File: 1-two-sum/two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> hmap = new HashMap<>();
        int ans[] = new int[2];

        for(int i = 0; i<nums.length; i++){
            if(hmap.containsKey(target - nums[i])){
                
                ans[0] = hmap.get(target-nums[i]);
                ans[1] = i;
                return ans;
            }

            hmap.put(nums[i], i);
        }

        return ans;
    }
}

File: 1-two-sum/two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> map;
        vector<int> ans;

        for(int i = 0; i<nums.size(); i++){
            int first = nums[i];
            int second = target - first;

            if(map.find(second) != map.end()){
                ans.push_back(i);
                ans.push_back(map[second]);
                break;
            }

            map[first] = i;
        }

        return ans;
    }
};